webapplication/user/models.py

```diff
@@ -114,8 +114,12 @@ class Meta:
         )
 
     @staticmethod
-    def generate_comment(request: Request, aoi: AoI):
+    def generate_comment(request: Request, error=None):
         comment = f"{request.component_name}."
-        if aoi:
-            comment += f" Area: {aoi.name} processed."
+        if request.aoi:
+            comment += f" Area: {request.aoi.name}."
+        if not error:
+            comment += " Processing."
+        else:
+            comment += f" Error: '{error}'."
```

Comment:
I'd better (1) not materialize this at the transaction creation time, (2) move this logic to UI, transaction API should return necessary data about request such as its AOI (BTW it should always be set in request), corresponding errors (error ids).

Reply:
partly fixed